Physics driver with Muphys (#1301)
C2SM/icon4py: model/atmosphere/subgrid_scale_physics/physics_interface/README.md

# icon4py-atmosphere-physics-interface

> The physics orchestrator (`PhysicsDriver`) and its process registry (`PhysicsProcess`, `ProcessTimeControl`, `ForcingMode`) for icon4py, modelled on ICON-MPIM's AES physics interface (`aes_phy_main`).

Reviewer: ```suggestion
The physics driver (`PhysicsDriver`) and its process registry (`PhysicsProcess`, `ProcessTimeControl`, `ForcingMode`) for ICON4Py, modelled on ICON-MPIM's AES physics interface (`aes_phy_main`).
```

> The physics orchestrator (`PhysicsDriver`) and its process registry (`PhysicsProcess`, `ProcessTimeControl`, `ForcingMode`) for icon4py, modelled on ICON-MPIM's AES physics interface (`aes_phy_main`).

Reviewer 2: This may deserve a bit of bikeshedding:
- Since we're creating a new package, we should be relatively sure about the name. A plain `physics` is in my opinion also an option for the name, or `physics-driver` as the top level entry point of the package. I don't dislike `physics-interface`, but want to make sure we're not just going with the name because noone thought about alternatives.
- "its process registry" I don't think there's a process registry in the sense that I think of about a "registry" (a thing where other things can be registered and the first thing will know about all the registered things). The PhysicsProcess class also says "A registered physics process" which I don't think is useful terminology in this context. It can likely just say "A physics process: ..."? My suggestion would be

```suggestion
# icon4py-atmosphere-physics-interface

The physics driver (`PhysicsDriver`) and utilities (`PhysicsProcess`, `ProcessTimeControl`, `ForcingMode`), modelled on ICON-MPIM's AES physics interface (`aes_phy_main`).
```

I've also dropped the "for icon4py". This is an icon4py package, so I don't see for what else it should be.

> The L2 physics orchestrator (`PhysicsDriver`) and its process registry (`PhysicsProcess`, `ProcessTimeControl`, `ForcingMode`) for icon4py, modelled on ICON-MPIM's AES physics interface (`aes_phy_main`).

Reviewer: let's not use two or more names for the same thing ;-)
see discussion in #icon4py-dev

Reviewer 2: Also, this is still referencing the L2 naming. Please check if there are other places that need to have the LN naming changed.
